From 2e980a57c261407795511d4794c8d22e78ecb9b2 Mon Sep 17 00:00:00 2001
From: Rudi <r.ihle@s-t.de>
Date: Mon, 4 Jan 2016 08:41:22 +0100
Subject: [PATCH] video: mxc_hdmi: Insert standard VGA mode only once

Skip standard VGA if found in EDID. We force this mode to be present
by default and therefore should not add it a second time.

Signed-off-by: Rudi <r.ihle@s-t.de>
---
 drivers/video/mxc/mxc_hdmi.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/video/mxc/mxc_hdmi.c b/drivers/video/mxc/mxc_hdmi.c
index 182ae2a..ec4a438 100644
--- a/drivers/video/mxc/mxc_hdmi.c
+++ b/drivers/video/mxc/mxc_hdmi.c
@@ -95,7 +95,7 @@
  * in HDMI Initialization Step B
  */
 static const struct fb_videomode vga_mode = {
-	/* 640x480 @ 60 Hz, 31.5 kHz hsync */
+	/* 640x480 @ 59.94 Hz, 31.5 kHz hsync */
 	NULL, 60, 640, 480, 39721, 48, 16, 33, 10, 96, 2, 0,
 	FB_VMODE_NONINTERLACED | FB_VMODE_ASPECT_4_3, FB_MODE_IS_VESA,
 };
@@ -1988,6 +1988,10 @@ static void mxc_hdmi_create_modelist(struct mxc_hdmi *hdmi, int from_edid)
 				continue;
 		}
 
+		/* Skip standard VGA (already present) */
+		if (vic == 1 && (mode.flag & FB_MODE_IS_VESA))
+			continue;
+
 		/* TODO: Discuss if we should always set default modes as standard */
 		if (!from_edid /*&& ignore_edid*/)
 			mode.flag |= FB_MODE_IS_STANDARD;
-- 
1.9.1

